app/routes/announcements/components/AnnouncementsCreate.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+import React from 'react';
+import styles from './AnnouncementsList.css';
+import Flex from 'app/components/Layout/Flex';
+import { Form, SelectInput, TextArea } from 'app/components/Form';
+import { reduxForm, Field, reset } from 'redux-form';
+import Button from 'app/components/Button';
+import { ContentMain } from 'app/components/Content';
+
+type Props = {
+  createAnnouncement: (...any) => void,
+  actionGrant: Array<string>,
+  handleSubmit: Function => void,
+  invalid: boolean,
+  pristine: boolean,
+  submitting: boolean
+};
+
+const AnnouncementsCreate = ({
+  createAnnouncement,
+  actionGrant,
+  handleSubmit,
+  invalid,
+  pristine,
+  submitting
+}: Props) => {
+  const disabledButton = invalid || pristine || submitting;
+
+  const onSubmit = (announcement, send = false) => {
+    return createAnnouncement({
+      ...announcement,
+      users: announcement.users
+        ? announcement.users.map(user => user.value)
+        : [],
+      groups: announcement.groups
```

Comment:
You could probably avoid all these checks if you passed AnnouncementCreate some `initialValues`:

```
const AnnouncementsCreate = ...;

const initialValues = {
    events: [],
    meetings: [],
    ...
}

export default reduxForm({
   initialValues,
   ...
})(AnnouncementCreate);
```